app/controllers/tickets_controller.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+class TicketsController < ApplicationController
+  before_action :set_ticket, only: %i[show update]
+  before_action :set_contact, only: %i[update]
+  before_action :set_participating_seller, only: %i[create]
+
+  # GET /tickets
+  def index
+    @tickets = Ticket.all
+
+    json_response(@tickets)
+  end
+
+  # GET /tickets/1
+  def show
+    json_response(@ticket)
+  end
+
+  # POST /tickets
+  def create
+    savedCorrectly = true
+    createdTickets = []
+    numTix = params[:number_of_tickets]
+
+    if numTix.present? && numTix > 0
+      ActiveRecord::Base.transaction do
+        (1..numTix).each do
+          attributes = {}
+          attributes[:ticket_id] = Ticket.generate_ticket_id
+          attributes[:participating_seller] = @participating_seller
+
+          @ticket = Ticket.new(attributes)
+          savedCorrectly &&= @ticket.save!
+          createdTickets << @ticket
+        end
+      end
+    else
+      savedCorrectly = false
+    end
+
+    if savedCorrectly
+      json_response(createdTickets, :created)
+    else
+      json_response({ message: 'Error: malformed request, expecting participating_seller_id and number_of_tickets' }, :unprocessable_entity)
```

Comment:
Instead you can do 

`raise InvalidParameterError, 'Error: malformed request, expecting participating_seller_id and number_of_tickets'`

and it will be caught by our exception_handller

Reply:
Makes sense to me, switched it all and it works a lot better and is a TON cleaner this way thanks 👍